packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -587,54 +587,54 @@ export class FluidDataStoreRuntime
 		return this.dataStoreContext.uploadBlob(blob, signal);
 	}
 
+	private createRemoteChannelContext(message) {
+		const attachMessage = (message.contents ?? message.content) as IAttachMessage;
```

Comment:
I think we can reduce the number of unsafe casts in this code by having this method take `attachMessage: IAttachMessage` and having the consumer do the cast. this also keep this code isolated from the inference we do to figure out the message type